loopy/statistics.py

```diff
@@ -1670,24 +1670,27 @@ def _get_op_map_for_single_kernel(knl, callables_table,
     op_map = op_counter.new_zero_poly_map()
 
     from loopy.kernel.instruction import (
-            CallInstruction, CInstruction, Assignment,
+            MultiAssignmentBase, CInstruction,
             NoOpInstruction, BarrierInstruction)
 
     for insn in knl.instructions:
-        if isinstance(insn, (CallInstruction, CInstruction, Assignment)):
-            ops = op_counter(insn.assignees) + op_counter(insn.expression)
-            for key, val in ops.count_map.items():
-                count = _get_insn_count(knl, callables_table, insn.id,
-                            subgroup_size, count_redundant_work,
-                            key.count_granularity)
-                op_map = op_map + ToCountMap({key: val}) * count
+        if isinstance(insn, MultiAssignmentBase):
+            exprs_in_insn = (insn.assignees, insn.expression,
+                             tuple(insn.predicates))
 
-        elif isinstance(insn, (NoOpInstruction, BarrierInstruction)):
-            pass
+        elif isinstance(insn, (CInstruction, NoOpInstruction, BarrierInstruction)):
+            exprs_in_insn = tuple(insn.predicates)
```

Comment:
Since we don't know how much computation goes on in a `CInstruction`, I think we at least need a flag to turn on this behavior, since it's easy to have misconceptions on what it does.

Reply:
I added a warning in https://github.com/inducer/loopy/pull/649